frontends/p4/evaluator/evaluator.h

```diff
@@ -41,7 +41,7 @@ class Evaluator final : public Inspector, public IHasBlock {
  public:
     Evaluator(const ReferenceMap* refMap, const TypeMap* typeMap) :
             refMap(refMap), typeMap(typeMap), toplevelBlock(nullptr)
-    { CHECK_NULL(refMap); CHECK_NULL(typeMap); setName("Evaluator"); }
+    { CHECK_NULL(refMap); CHECK_NULL(typeMap); setName("Evaluator"); visitDagOnce = false; }
```

Comment:
This change leads to a bunch of crashes in `setValue` -- by visiting the same node multiple times, it tries to call setValue multiple times, leading to the BUG_CHECK at ir/ir.cpp:194.  Why was it added?

Reply:
The test case that this PR fixes is a good example.
Each block has a map of constant values as evaluated within that block.
If you have a diamond pattern, where two blocks instantiate a third one, some constants are evaluated only once if you don't do visitDagOnce = false, so they are missing from some maps.
This is the right way to do this. 

If you have a test case which crashes please share it.